controllers/managedocs_controller.go

```diff
@@ -1067,6 +1085,52 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getPrometheusRemoteWriteSpec() (*promv1.RemoteWriteSpec, error) {
+	if err := r.get(r.rhobsSecret); err != nil {
+		return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+	}
+	rhobsSecretData := r.rhobsSecret.Data
+	rhobsUserID := rhobsSecretData[rhobsClientId]
+	if rhobsUserID == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", rhobsClientId)
+	}
```

Comment:
A couple of notes:
- I think naming here is very confusing, and we already have a convention for secret keys (`*SecretKey`)
- You do not use the value so having a var for the value can be avoided
- the key is used only once and only here 

```suggestion
	if _ , found := rhobsSecretData["prom-remote-write-config-id"]; !found {
		return nil, fmt.Error("rhobs secret does not contain a value for key prom-remote-write-config-id")
	}
```

Reply:
Don